### PR TITLE
fix(zone-configuration): add images to IDE doc, extend code docs

### PR DESCRIPTION
**Pull request by BioPhoton**
This PR includes:

- I moved the documentation for top-level APIs into the types file
- Added docs for runtime settings
- Added assertion for runtime settings
- Added tests
- Added docs and images for IDE autocomplete and inline docs

### CODE REVIEW

File: libs/cdk/docs/zone-configuration/how-to-setup-zone-flags.md [hunk 1]
```diff
@@ -103,3 +119,26 @@ import 'zone.js/dist/zone';
...
+```typescript
+// Zone JS is required by default for Angular itself.
+import 'zone.js/dist/zone';
+// ☝️ Make sure zoneflags are imported before zone.js
+import './zone-runtime';
+```
+
+> **💡 Pro Tipps:** > `@rx-angular/cdk/zone-configuration` errors if it is used incorrectly.
+> If you used zone-flags wrong (not executing it before zone.js runs) you should see the following error in the console:
+> ![Log zone-flags](https://raw.githubusercontent.com/rx-angular/rx-angular/master/libs/cdk/docs/zone-configuration/images/angular-zone-flags_log-zone-flags-runtime_michael-hladky.png)
```
Comment (BioPhoton) [inline libs/cdk/docs/zone-configuration/how-to-setup-zone-flags.md:144]:
```suggestion
> If you used zone-runtime configurations wrong (not executing it after zone.js runs) you should see the following error in the console:
> ![Log zone-runtime](https://raw.githubusercontent.com/rx-angular/rx-angular/master/libs/cdk/docs/zone-configuration/images/angular-zone-flags_log-zone-flags-runtime_michael-hladky.png)
```

File: libs/cdk/docs/zone-configuration/how-to-setup-zone-flags.md [hunk 2]
```diff
@@ -82,14 +89,23 @@ import 'zone.js/dist/zone';
 1. Create file `zone-flags.ts` parallel to your `polyfills.ts` and insert the following content:
```
Comment (hoebbelsB) [inline libs/cdk/docs/zone-configuration/how-to-setup-zone-flags.md:89]:
```suggestion
1. Create file `zone-flags.ts` next to your `polyfills.ts` and insert the following content:
```

File: libs/cdk/src/lib/zone-configuration/model/zone-cfg.types.ts [hunk 3]
```diff
@@ -43,15 +43,44 @@ export interface ZoneConfigConvenienceMethods {
 }
 
 export interface ZoneConfigConfiguration {
+  /**
+   * Interface of `zone.js` configurations.
+   *
+   * You can define the following configurations on the `window/global` object before
+   * importing `zone.js` to change `zone.js` default behaviors.
```
Comment (hoebbelsB) [inline libs/cdk/src/lib/zone-configuration/model/zone-cfg.types.ts:50]:
```suggestion
   * importing `zone.js` to change its default behaviors.
```

File: libs/cdk/src/lib/zone-configuration/model/zone-cfg.types.ts [hunk 4]
```diff
@@ -43,15 +43,44 @@ export interface ZoneConfigConvenienceMethods {
...
+   */
   global: {
     disable: GlobalDisableConfigurationMethods;
   };
+  /**
+   * Interface of `zone-testing.js` test configurations.
+   *
+   * You can define the following configurations on the `window` or `global` object before
+   * importing `zone-testing.js` to change `zone-testing.js` default behaviors in the test runner.
```
Comment (hoebbelsB) [inline libs/cdk/src/lib/zone-configuration/model/zone-cfg.types.ts:59]:
```suggestion
   * importing `zone-testing.js` to change its default behaviors in the test runner.
```

File: libs/cdk/src/lib/zone-configuration/zone-cfg.ts [hunk 5]
```diff
@@ -25,30 +25,49 @@ const zoneSymbol = '__zone_symbol__';
 /**
  * https://angular.io/guide/zone#setting-up-zonejs
  **/
-function assertZoneConfig() {
+function assertZoneObjectPresentBeforeZoneJs() {
```
Comment (hoebbelsB) [inline libs/cdk/src/lib/zone-configuration/zone-cfg.ts:28]:
```suggestion
function assertZoneObjectExistsBeforeZoneJs() {
```

be careful with this one please :D did not make sure that usages are refactored as well

**Review by hoebbelsB — CHANGES_REQUESTED** [review]
very nice! added some minor improvements for js docs

File: libs/cdk/src/lib/zone-configuration/zone-cfg.ts (hunk 5 above)
Comment (BioPhoton) [reply libs/cdk/src/lib/zone-configuration/zone-cfg.ts:28]: done

File: libs/cdk/docs/zone-configuration/how-to-setup-zone-flags.md [hunk 6]
```diff
@@ -103,3 +119,26 @@ import 'zone.js/dist/zone';
 > **💡 Pro Tipps:** > `@rx-angular/cdk/zone-configuration` errors if it is used incorrectly.
```
Comment (hoebbelsB) [inline libs/cdk/docs/zone-configuration/how-to-setup-zone-flags.md:119]:
```suggestion
> **💡 Pro Tip:** > `@rx-angular/cdk/zone-configuration` errors if it is used incorrectly.
```

File: libs/cdk/docs/zone-configuration/how-to-setup-zone-flags.md [hunk 7]
```diff
@@ -103,3 +119,26 @@ import 'zone.js/dist/zone';
...
+2. In `polyfills.ts` below the zone import, import `zone-runtime.ts`:
+
+```typescript
+// Zone JS is required by default for Angular itself.
+import 'zone.js/dist/zone';
+// ☝️ Make sure zoneflags are imported before zone.js
+import './zone-runtime';
+```
+
+> **💡 Pro Tipps:** > `@rx-angular/cdk/zone-configuration` errors if it is used incorrectly.
```
Comment (hoebbelsB) [inline libs/cdk/docs/zone-configuration/how-to-setup-zone-flags.md:142]:
```suggestion
> **💡 Pro Tip:** > `@rx-angular/cdk/zone-configuration` errors if it is used incorrectly.
```

**Review by hoebbelsB — CHANGES_REQUESTED** [review]
another typo found :D sorry

**Review by hoebbelsB — APPROVED** [review]

**Pull request merged**